docs/examples/workflows/callable_script.md

```diff
@@ -0,0 +1,150 @@
+
+
+    # Note that the input type is a list of Input
+    # hera can also automatically de-serialize
+    # composite types like lists and dicts
+    @script()
+    def another_function(inputs: List[Input]) -> Output:
+        return Output(output=inputs)
+
+
+    # it also works with raw json strings
+    # but those must be explicitly marked as
+    # a string type
+    @script()
+    def str_function(input: str) -> Output:
+        # Example function to ensure we are not json parsing
+        # string types before passing it to the function
+        return Output(output=[Input.parse_raw(input)])
+
+
+    with Workflow(name="my-workflow") as w:
+        with Steps(name="my-steps") as s:
+            my_function(arguments={"input": Input(a=2, b="bar")})
+            str_function(arguments={"input": Input(a=2, b="bar").json()})
+            another_function(arguments={"inputs": [Input(a=2, b="bar"), Input(a=2, b="bar")]})
+    ```
+
+=== "YAML"
+
+    ```yaml linenums="1"
+    apiVersion: argoproj.io/v1alpha1
+    kind: Workflow
+    metadata:
+      name: my-workflow
+    spec:
+      templates:
+      - name: my-steps
+        steps:
+        - - arguments:
+              parameters:
+              - name: input
+                value: '{"a": 2, "b": "bar"}'
+            name: my-function
+            template: my-function
+        - - arguments:
+              parameters:
+              - name: input
+                value: '{"a": 2, "b": "bar"}'
+            name: str-function
+            template: str-function
+        - - arguments:
+              parameters:
+              - name: inputs
+                value: '[{"a": 2, "b": "bar"}, {"a": 2, "b": "bar"}]'
+            name: another-function
+            template: another-function
+      - inputs:
+          parameters:
+          - name: input
+        name: my-function
+        script:
+          args:
+          - -m
+          - hera.workflows.runner
+          - -e
+          - examples.workflows.callable_script:my_function
+          command:
+          - python
+          image: my-image-with-python-source-code-and-dependencies
+          source: '{{inputs.parameters}}'
+      - inputs:
+          parameters:
+          - name: input
+        name: str-function
+        script:
+          args:
+          - -m
+          - hera.workflows.runner
+          - -e
+          - examples.workflows.callable_script:str_function
+          command:
+          - python
+          image: my-image-with-python-source-code-and-dependencies
+          source: '{{inputs.parameters}}'
+      - inputs:
+          parameters:
+          - name: inputs
+        name: another-function
+        script:
+          args:
+          - -m
+          - hera.workflows.runner
+          - -e
+          - examples.workflows.callable_script:another_function
+          command:
+          - python
+          image: my-image-with-python-source-code-and-dependencies
+          source: '{{inputs.parameters}}'
```

Comment:
We put the input parameters in a source file and then parse it as json to avoid python string interpolation issues with quotes or brackets. This also leads to cleaner yamls which are readable.